src/common/eeprom.cpp

```diff
@@ -509,3 +595,132 @@ int8_t eeprom_test_PUT(const unsigned int bytes) {
     }
     return res_flag;
 }
+
+///////////////////////////////////////////////////////////////////////////////
+/// Sheets profile methods
+uint32_t sheet_next_calibrated() {
+#if (EEPROM_FEATURES & EEPROM_FEATURE_SHEETS)
+    uint8_t index = variant_get_ui8(eeprom_get_var(EEVAR_ACTUAL_SHEET));
+
+    for (int8_t i = 1; i < MAX_SHEETS; ++i) {
+        if (sheet_select((index + i) % MAX_SHEETS))
+            return (index + i) % MAX_SHEETS;
+    }
+#endif
+    return 0;
+}
+
+bool sheet_is_calibrated(uint32_t index) {
+#if (EEPROM_FEATURES & EEPROM_FEATURE_SHEETS)
+    uint16_t profile_address = eeprom_var_addr(EEVAR_SHEET_PROFILE0 + index);
+    float z_offset = FLT_MAX;
+    st25dv64k_user_read_bytes(profile_address + MAX_SHEET_NAME_LENGTH,
+        &z_offset, sizeof(float));
+    return !nearlyEqual(z_offset, FLT_MAX, 0.001f);
+#else
+    return index == 0;
+#endif
+}
+
+bool sheet_select(uint32_t index) {
+#if (EEPROM_FEATURES & EEPROM_FEATURE_SHEETS)
+    if (index >= MAX_SHEETS || !sheet_is_calibrated(index))
+        return false;
+    uint16_t actual_sheet_address = eeprom_var_addr(EEVAR_ACTUAL_SHEET);
+    st25dv64k_user_write(actual_sheet_address, index);
+    return true;
+#else
+    return index == 0;
+#endif
+}
+
+bool sheet_calibrate(uint32_t index) {
+#if (EEPROM_FEATURES & EEPROM_FEATURE_SHEETS)
+    if (index >= MAX_SHEETS)
+        return false;
+    uint16_t actual_sheet_address = eeprom_var_addr(EEVAR_ACTUAL_SHEET);
+    st25dv64k_user_write(actual_sheet_address, index);
+    return true;
+#else
+    return index == 0;
+#endif
+}
+
+bool sheet_reset(uint32_t index) {
+#if (EEPROM_FEATURES & EEPROM_FEATURE_SHEETS)
+    if (index >= MAX_SHEETS)
+        return false;
+    uint8_t actual = variant_get_ui8(eeprom_get_var(EEVAR_ACTUAL_SHEET));
+    uint16_t profile_address = eeprom_var_addr(EEVAR_SHEET_PROFILE0 + index);
+    float z_offset = FLT_MAX;
+    st25dv64k_user_write_bytes(profile_address + MAX_SHEET_NAME_LENGTH,
```

Comment:
do you reset the name of the sheet as well somewhere?

Reply:
Not yet. When the renaming is not implemented in the GUI, such a feature has no reason yet.

Reply:
ok